src/containers/IdeLoader.tsx

```diff
@@ -0,0 +1,190 @@
+      this.loadFactoryPageCallbacks.showAlert(AlertVariant.danger, message);
+    } else {
+      console.error(message);
+    }
+  }
+
+  public async componentDidMount(): Promise<void> {
+    const { allWorkspaces, requestWorkspaces } = this.props;
+    if (!allWorkspaces || allWorkspaces.length === 0) {
+      await requestWorkspaces();
+    }
+
+    await this.initWorkspace();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    const { allWorkspaces, match: { params } } = this.props;
+    const { currentStep } = this.state;
+    const workspace = allWorkspaces.find(workspace =>
+      workspace.namespace === params.namespace && workspace.devfile.metadata.name === params.workspaceName);
+    if (workspace && WorkspaceStatus[workspace.status] === WorkspaceStatus.ERROR) {
+      this.setState({
+        currentStep,
+        hasError: true,
+      });
+      return;
+    }
+    await this.initWorkspace();
+  }
+
+  private async openIDE(workspaceId: string): Promise<void> {
+    await this.props.requestWorkspace(workspaceId);
+    const workspace = this.props.allWorkspaces.find(workspace =>
+      workspace.id === workspaceId);
+    if (!workspace || !workspace.runtime) {
+      return;
+    }
+    let ideUrl = '';
+    const machines = workspace.runtime.machines || {};
+    for (const machineName of Object.keys(machines)) {
+      const servers = machines[machineName].servers || {};
+      for (const serverId of Object.keys(servers)) {
+        const attributes = (servers[serverId] as any).attributes;
+        if (attributes && attributes['type'] === 'ide') {
+          ideUrl = servers[serverId].url;
+          break;
+        }
+      }
+    }
+    if (!ideUrl) {
+      this.showErrorAlert('Don\'t know what to open, IDE url is not defined.');
+      return;
+    }
+    await timeout();
+    this.setState({ currentStep: 4, ideUrl });
+  }
+
+  private async initWorkspace(): Promise<void> {
+    const { allWorkspaces, match: { params } } = this.props;
+    const { namespace, workspaceName } = this.state;
+
+    if (namespace !== params.namespace || workspaceName !== params.workspaceName) {
+      this.setState({ currentStep: 1, hasError: false, ideUrl: '', namespace, workspaceName });
+      await timeout();
+      return;
+    } else if (this.state.currentStep > 2) {
+      return;
+    }
+    const workspace = allWorkspaces.find(workspace =>
+      workspace.namespace === params.namespace && workspace.devfile.metadata.name === params.workspaceName);
+    if (workspace) {
+      this.setState({ currentStep: this.state.currentStep, workspaceId: workspace.id });
+      if (this.state.currentStep === 2 && WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING) {
+        this.setState({ currentStep: 3 });
+        return this.openIDE(workspace.id);
+      }
+    } else {
+      this.showErrorAlert('Failed to find the target workspace.');
+      return;
+    }
+    await timeout();
```

Comment:
As far as I understand you use `timeout()` to add a delays in the loading flow. Personally, I don't like this approach because I can't definitely say that 1000ms would be enough to resolve all kind of issues that may happen during a factory acceptance or workspace creation, when the flow could be continued. My question is have you considered any another approach? Maybe we need to request improvements on che-server side in order to provide necessary events?

cc @sleshchenko

Reply:
Discussed about `timeout()` service with @olexii4 . He explained, that this service is only used to improve IDE loading steps animation and doesn't affect the loading flow itself.

Reply:
It is a minimal delay for layout wizard steps.
I will change to 0.5s. It is a good solution for the initial implementation of this flow.

Reply:
This delay is only needed for the wizard(visual effects only).